samza-sql/src/main/java/org/apache/samza/sql/runner/SamzaSqlApplicationRunner.java

```diff
@@ -0,0 +1,134 @@
+ * otherwise directly calls the RemoteApplicationRunner which automatically performs the config rewriting .
+ */
+public class SamzaSqlApplicationRunner extends AbstractApplicationRunner {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SamzaSqlApplicationRunner.class);
+
+  private final Config sqlConfig;
+  private final ApplicationRunner appRunner;
+  private final Boolean localRunner;
+
+  public static final String RUNNER_CONFIG = "app.runner.class";
+  public static final String CFG_FMT_SAMZA_STREAM_SYSTEM = "streams.%s.samza.system";
+
+  public SamzaSqlApplicationRunner(Config config) {
+    this(false, config);
+  }
+
+  public SamzaSqlApplicationRunner(Boolean localRunner, Config config) {
+    super(config);
+    this.localRunner = localRunner;
+    sqlConfig = computeSamzaConfigs(localRunner, config);
+    appRunner = ApplicationRunner.fromConfig(sqlConfig);
+  }
+
+  public static Config computeSamzaConfigs(Boolean localRunner, Config config) {
+    Map<String, String> newConfig = new HashMap<>();
+
+    SourceResolver sourceResolver = SamzaSqlApplicationConfig.createSourceResolver(config);
+    // Parse the sql and find the input stream streams
+    List<String> sqlStmts = SamzaSqlApplicationConfig.fetchSqlFromConfig(config);
+
+    // This is needed because the SQL file may not be available in all the node managers.
+    String sqlJson = SamzaSqlApplicationConfig.serializeSqlStmts(sqlStmts);
+    newConfig.put(SamzaSqlApplicationConfig.CFG_SQL_STMTS_JSON, sqlJson);
+
+    List<SamzaSqlQueryParser.QueryInfo> queryInfo = SamzaSqlApplicationConfig.fetchQueryInfo(sqlStmts);
+    for (SamzaSqlQueryParser.QueryInfo query : queryInfo) {
+      // Populate stream to system mapping config for input and output system streams
+      for (String inputSource : query.getInputSources()) {
+        SystemStreamConfig inputSystemStreamConfig = sourceResolver.fetchSourceInfo(inputSource);
+        newConfig.put(String.format(CFG_FMT_SAMZA_STREAM_SYSTEM, inputSystemStreamConfig.getStreamName()),
+            inputSystemStreamConfig.getSystemName());
+      }
+      SystemStreamConfig outputSystemStreamConfig = sourceResolver.fetchSourceInfo(query.getOutputSource());
+      newConfig.put(String.format(CFG_FMT_SAMZA_STREAM_SYSTEM, outputSystemStreamConfig.getStreamName()),
+          outputSystemStreamConfig.getSystemName());
+    }
+
+    if (localRunner) {
+      newConfig.put(RUNNER_CONFIG, LocalApplicationRunner.class.getName());
+    } else {
+      newConfig.put(RUNNER_CONFIG, RemoteApplicationRunner.class.getName());
+    }
+
+    newConfig.putAll(config);
+
+    LOG.info("New Samza configs: " + newConfig);
+    return new MapConfig(newConfig);
+  }
+
+  public void runAndWaitForFinish() {
+    Validate.isTrue(localRunner, "This method can be called only in standalone mode.");
+    SamzaSqlApplication app = new SamzaSqlApplication();
+    run(app);
+    ((LocalApplicationRunner) appRunner).waitForFinish();
```

Comment:
This might result in exception when localRunner == false.

Reply:
Yes! that is intended. We don't want this method to be called by Remote runners.